### PR TITLE
[add] Dashboard Private Route

### PR DESCRIPTION
**Pull request by edbella**
Created a dashboard private route that is only accessible by an authenticated user/vendor.

### CODE REVIEW

File: src/components/dashboard/components/Dashboard.jsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import { useSelector } from 'react-redux';
+import UserDashboard from './UserDashboard';
+import VendorDashboard from './VendorDashboard';
+
+export function Dashboard() {
+    const appState = useSelector(state => state);
+
+    const { signIn: { isAuthenticated, userObj } } = appState;
+
+    const { userType } = userObj;
```
Comment (Way2nnadi) [inline src/components/dashboard/components/Dashboard.jsx]: no spaces between `const`s, treat it like a block of code. spaces around `if/else`

File: src/components/dashboard/components/VendorDashboard.jsx [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import React from 'react';
+
+export default function VendorDashboard() {
+    return (
+        <div />
+    );
+}
```
Comment (Way2nnadi) [inline src/components/dashboard/components/VendorDashboard.jsx]: This is not doing anything

File: src/components/dashboard/components/Dashboard.jsx (hunk 1 above)
Comment (edbella) [reply src/components/dashboard/components/Dashboard.jsx]: Ok

File: src/components/dashboard/components/VendorDashboard.jsx (hunk 2 above)
Comment (edbella) [reply src/components/dashboard/components/VendorDashboard.jsx]: Yes. Its the next component to be built but for the sake of this PR I just wanted to focus on creating the protected route solely.

**Pull request merged**